CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Unfortunately, we do. `add_compile_options` and the particular syntax of `add_executable` used, for example, don't work with lower versions. This could potentially be ported, but it is more effort than simply lowering the version number and loses some clarity in the process.

Reply:
Downloaded CMake 3.5.1, and indeed you are right.

On the other hand, Ubuntu 18.04 has only CMake 3.10. Given that a few years ago we had players with 14.04, I'm afraid we will have questions about how to install CMake. This is not a stopper for merging the changes of course, but I wonder how do you feel in general about backporting it to 3.5 in future.

Reply:
I'm a bit conflicted. On the one hand, I do understand the argument and it would be nice if it worked with the distro-shipped versions. On the other hand, newer CMake versions do add useful new features, and having to restrict yourself to outdated dev tools is painful. I'm not very happy with the Debian/Ubuntu conservative update policy on dev tools, in general :)

The question for me is how many users will actually attempt to compile TDM who do not have the technical expertise to install CMake manually. I'm inclined to wait and see if there's an actual demand before investing time in a backport, because I think that time could be better spent at the moment.

Reply:
BTW, CMake is available as a Snap package, and given that Ubuntu seems to push Snap packages in general, that may lessen the importance of the APT version?

Reply:
Yes, snap is a good idea. I should try it out =)
Let's forget about this question then.